go-controller/pkg/ovn/egressgw.go

```diff
@@ -967,6 +951,19 @@ func (oc *Controller) cleanExGwECMPRoutes() {
 		klog.Infof("Syncing exgw routes took %v", time.Since(start))
 	}()
 
+	// migration from LGW to SGW mode
+	// for shared gateway mode, these LRPs shouldn't exist, so delete them all
+	if config.Gateway.Mode == config.GatewayModeShared {
+		if err := oc.delAllHybridRoutePolicies(); err != nil {
```

Comment:
`delAllHybridRoutePolicies` includes legacy as well, so we should be good for SGW and with adding the `else if` for LGW and doing only legacy removal for that.